packages/react/src/components/Table/index.test.tsx

```diff
@@ -0,0 +1,314 @@
+      </TableHead>
+    </Table>
+  );
+
+  expect(screen.getByRole('button')).toBeInTheDocument();
+  expect(screen.getByRole('status').closest('.Icon--sort-triangle'));
+  expect(screen.getByRole('status')).toHaveTextContent('');
+});
+
+test('should render className "TableHeader--sorting" when actively sorting', () => {
+  render(
+    <Table>
+      <TableHead>
+        <TableRow>
+          <TableHeader sortDirection={'ascending'} onSort={() => null}>
+            Sortable Header
+          </TableHeader>
+        </TableRow>
+      </TableHead>
+    </Table>
+  );
+
+  expect(screen.getByRole('status').closest('.TableHeader--sort-ascending'));
+});
+
+test('should render triangle up Icon and ascending message when sortDirection is ascending', () => {
+  render(
+    <Table>
+      <TableHead>
+        <TableRow>
+          <TableHeader
+            sortDirection={'ascending'}
+            sortAscendingAnnouncement={'up and away'}
+            onSort={() => null}
+          >
+            Sortable Header
+          </TableHeader>
+        </TableRow>
+      </TableHead>
+    </Table>
+  );
+
+  expect(screen.getByText('up and away')).toBeInTheDocument();
+  expect(screen.getByRole('status').closest('.Icon--table-sort-ascending'));
+});
+
+test('should render triangle down Icon and descending message when sortDirection is descending', () => {
+  render(
+    <Table>
+      <TableHead>
+        <TableRow>
+          <TableHeader
+            sortDirection={'descending'}
+            sortDescendingAnnouncement={'down below'}
+            onSort={() => null}
+          >
+            Sortable Header
+          </TableHeader>
+        </TableRow>
+      </TableHead>
+    </Table>
+  );
+
+  expect(screen.getByText('down below')).toBeInTheDocument();
+  expect(screen.getByRole('status').closest('.Icon--table-sort-descending'));
+});
+
+test('should call onSort when sort button is clicked', async () => {
+  const onSortMock = jest.fn();
+
+  render(
+    <Table>
+      <TableHead>
+        <TableRow>
+          <TableHeader sortDirection="none" onSort={onSortMock}>
+            Sortable Header
+          </TableHeader>
+        </TableRow>
+      </TableHead>
+    </Table>
+  );
+
+  const button = screen.getByRole('button');
+
+  await userEvent.click(button);
+  await waitFor(() => {
+    expect(onSortMock).toHaveBeenCalledTimes(1);
+  });
+});
+
+test('should maintain focus on the sort button after it is clicked', async () => {
+  render(
+    <Table>
+      <TableHead>
+        <TableRow>
+          <TableHeader
+            id="button-focused"
+            sortDirection="none"
+            onSort={() => null}
+          >
+            Sortable Header
+          </TableHeader>
+        </TableRow>
+      </TableHead>
+    </Table>
+  );
+
+  const button = screen.getByRole('button');
+
+  await userEvent.click(button);
+  await waitFor(() => {
+    expect(button).toHaveFocus();
+  });
+});
+
+test('returns 0 axe violations', async () => {
```

Comment:
This test looks good, we just need additional axe tests for sorted columns (none, ascending, descending).